rlp/rlpgen/gen.go

```diff
@@ -373,6 +373,29 @@ func (op uint256Op) genDecode(ctx *genContext) (string, string) {
 	return result, b.String()
 }
 
+func (bctx *buildContext) makeNamedBasicOp(named *types.Named) (op, error) {
+	underlying := named.Underlying()
+	basic, ok := underlying.(*types.Basic)
+	if !ok {
+		return nil, fmt.Errorf("expected basic type, got %T", underlying)
+	}
+
+	// Use the existing makeBasicOp function to get the base operation
+	baseOp, err := bctx.makeBasicOp(basic)
+	if err != nil {
+		return nil, err
+	}
+
+	// Cast to basicOp and modify the typ field to use the named type
+	op, ok := baseOp.(basicOp)
+	if !ok {
+		return nil, fmt.Errorf("expected basicOp, got %T", baseOp)
+	}
+	op.typ = named // Use the named type as the main type instead of the underlying basic type
```

Comment:
```suggestion
	// Overriding the type with the original will cause [basicOp.writeNeedsConversion] to
	// return true, resulting in the desired type conversion.
	op.typ = named
```

This is a "why" comment so can't be communicated with the code itself.